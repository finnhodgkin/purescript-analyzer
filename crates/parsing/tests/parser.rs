macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
