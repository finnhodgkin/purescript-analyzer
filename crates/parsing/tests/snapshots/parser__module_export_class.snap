---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..59
      ModuleHeader@0..59
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..25
          UPPER@7..24 "ModuleExportClass"
          Comment@24..25
            WHITESPACE@24..25 " "
        ModuleExportList@25..53
          LEFT_PARENTHESIS@25..26 "("
          ModuleExportClass@26..33
            CLASS@26..31 "class"
            Comment@31..32
              WHITESPACE@31..32 " "
            UPPER@32..33 "A"
          COMMA@33..34 ","
          Comment@34..35
            WHITESPACE@34..35 " "
          ModuleExportClass@35..41
            CLASS@35..40 "class"
            Comment@40..41
              WHITESPACE@40..41 " "
            ERROR@41..41 ""
          ERROR@41..41 ""
          ModuleExportValue@41..42
            LOWER@41..42 "b"
          COMMA@42..43 ","
          Comment@43..44
            WHITESPACE@43..44 " "
          ModuleExportClass@44..51
            CLASS@44..49 "class"
            Comment@49..50
              WHITESPACE@49..50 " "
            UPPER@50..51 "C"
          RIGHT_PARENTHESIS@51..52 ")"
          Comment@52..53
            WHITESPACE@52..53 " "
        WHERE@53..58 "where"
        Comment@58..59
          WHITESPACE@58..59 "\n"
      LAYOUT_START@59..59 ""
      ModuleImports@59..59
      ModuleStatements@59..59
      LAYOUT_END@59..59 ""
      END_OF_FILE@59..59 ""
    ,
    [
        ParseError {
            position: Position {
                offset: 41,
                line: 1,
                column: 42,
            },
            message: "expected UPPER",
        },
        ParseError {
            position: Position {
                offset: 41,
                line: 1,
                column: 42,
            },
            message: "expected COMMA",
        },
    ],
)
